src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(this.getClass())) {
+            Rectangle current = (Rectangle) rectangle;
+            return (this.width == current.width || this.width != null
+                    && this.width.equals(current.width))
+                    && (this.length == current.length || this.length != null
+                    && this.length.equals((current.length)))
+                    && (this.color == current.color || this.color != null
+                    && this.color.equals(current.color));
```

Comment:
we don't have to use `this` keyword here - it is redundant

Reply:
thanks
done